File: JavaScript/276-Paint Fence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {number}
 */
var numWays = function(n, k) {
    if (k === 0 || (n > 2 && k === 1) ){
        return 0;
    }
    
    let same = 0, diff = k;
    
    for (let i = 1; i < n; i++){
        let tmp = diff;
        diff = (diff + same)*(k-1);
        same = tmp;
    }
    
    return same + diff;
};

File: 276-Paint Fence.py
class Solution(object):
    def numWays(self, n, k):
        """
        :type n: int
        :type k: int
        :rtype: int
        """
        if k == 0 or (k==1 and n>2):
            return 0
        
        same, diff = 0, k
        
        for i in range(1, n):
            tmp = diff
            diff = (diff+same)*(k-1)
            same = tmp
            
        return same + diff